lib/screens/home/home_screen.dart

```diff
@@ -40,60 +51,71 @@ class _HomeScreenState extends State<HomeScreen> {
             arguments: _calendarController.selectedDay),
         child: Icon(Icons.add),
       ),
-      body: Column(
-        children: [
-          Material(
-            color: kPrimaryColor,
-            elevation: 3,
-            borderRadius: BorderRadius.only(
-              bottomRight: Radius.circular(10),
-              bottomLeft: Radius.circular(10),
-            ),
-            child: Container(
-              margin: EdgeInsets.only(
-                  left: kDefaultPadding, right: kDefaultPadding),
-              height: size.height * 0.3,
-              width: size.width,
-              decoration: BoxDecoration(
-                color: kPrimaryColor,
-                borderRadius: BorderRadius.only(
-                  bottomRight: Radius.circular(10),
-                  bottomLeft: Radius.circular(10),
-                ),
-              ),
-              child: Column(
-                crossAxisAlignment: CrossAxisAlignment.start,
-                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+      body: StreamBuilder<List<Wod>>(
+          stream: _wodsStream,
+          builder: (BuildContext context, AsyncSnapshot snapshot) {
+            if (snapshot.hasData) {
+              final List<Wod> wods = snapshot.data;
+              _groupWods(wods);
+              DateTime selectedDate = _calendarController.selectedDay;
+              List<Wod> _selectedWod = _data[selectedDate] ?? [];
+              void _onDaySelected(DateTime day, List wods, List holidays) {
```

Comment:
Ok, you are just creating a callback here to set the state of `_selectedWod`s and send it to `WodCalender`...

You already have a lot of variables to save state like `_selectedWod` (which is the list of WoDs to be shown on the screen for that date),  selected day and also selected Wod for the WodEditor screen.

When you have a lot of states to manage (particularly in different screens), it's when you need to use the Provider.
You can use the Provider at `HomeScreen` level with a `WodModel`  for instance where you can have your `currentWods` (list of Wods for the selected day to be shown on `HomeScreen`) and `selectedDay` and then you have access from the provider to any part of these states in `HomeScreen` or any screen that is inside this one.

I know it's a big change but you should do it while you can because this is already being hard to manage all those things. With a provider, your code will be better organized. If you can't do it, we can do it in the next session